### ISSUE TITLE
SNOW-285535: Implement callproc

### ISSUE THREAD

**Issue by whardier** [snowflakedb/snowflake-sqlalchemy#216]
Please implement callproc support.

Or....

Please demonstrate how to execute stored procedure calls without it.

Right now I have to use system calls to run cli tool.  Not secure or preferred.   Tried everything I could find on Github to make this work and came up short every time.

**Comment by sfc-gh-aling** [comment]
hey @whardier, apologize for keeping you waiting for so long and thanks for your feedback.

I want to clarify that `callproc` is not part of the sqlalchemy library, it's part of the [python dbapi2.0 spec(pep249)](https://peps.python.org/pep-0249/#callproc) which should be implemented in the connector library and sqlalchemy has documented [calling stored proceduer](https://docs.sqlalchemy.org/en/14/core/connections.html#calling-stored-procedures-and-user-defined-functions).

I just got a PR out for to [add callproc support to the snowflake-connector-python](https://github.com/snowflakedb/snowflake-connector-python/pull/1143) -- once it's merged and released, you shall be able to use the method.

----

In the meanwhile, instead of relying on the `callproc` method, you could achieve the same effect by calling `cursor.execute` method as long as you compose the sql in the correct form.

Here're some sample snippets that might be helpful to you:

```python
from sqlalchemy import create_engine
engine = create_engine(...)
connection = engine.raw_connection()
cursor = connection.cursor()

""" suppose you defined stored procedure as the following:
create or replace procedure my_sp (p1 string, p2 float)
returns string
language sql
as
$$
begin
    return 'done';
end;
$$

"""

res = cursor.execute("call stored_proceduer_name(%s, %s);", ('stringvalue', 1.23))
print(res.fetchall())  # it would print "[('done',)]"
```

By default the connector is using `format` [paramstyle](https://peps.python.org/pep-0249/#paramstyle), but we also support `pyformat`, `numeric` and `qmark` as the following code shows:

```python
import snowflake.connector
snowflake.connector.paramstyle = "pyformat"
r = engine.execute("call somefunc(%(v1)s);", {'v1': 'this py format'})
print(r.fetchall())

snowflake.connector.paramstyle = "numeric"
r = engine.execute("call somefunc(:1);", ('teststring',))
print(r.fetchall())

snowflake.connector.paramstyle = "qmark"
r = engine.execute("call somefunc(?);", ('teststring',))
print(r.fetchall())
```

----

I admit that our samples/docs need improvement, I will keep track and spend time improving them.
please let me know if there's anything we could help with.

**Comment by sfc-gh-aling** [comment]
FYI I've merged the PR for the callproc support, it will be included in our next `snowflake-connector-python` release.

**Comment by sfc-gh-aling** [comment]
we have released snowflake-connector-python 2.7.8 including the callproc support, please upgrade to the latest connector via `pip install snowflake-connector-python --upgrade` and try out the new feature.

I'm closing the issue now, thanks for your patience and please let us know if you need anything else.
